flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraCommitter.java

```diff
@@ -118,16 +117,32 @@ public void close() throws Exception {
 	}
 
 	@Override
-	public void commitCheckpoint(long checkpointID) {
-		session.execute(updateStatement.bind(checkpointID));
-		this.lastCommittedCheckpointID = checkpointID;
+	public void commitCheckpoint(int subtaskIdx, long checkpointId) {
+		String statement = String.format(
+			"UPDATE %s.%s set checkpoint_id=%d where sink_id='%s' and sub_id=%d;",
+			keySpace, table, checkpointId, operatorId, subtaskIdx);
+
+		session.execute(statement);
```

Comment:
You forgot to update the `lastCommittedCheckpointID` here. It will now only be updated once in `isCheckpointCommitted`.

As a follow up the test for this behavior should be extended to cover more than 1 commit.